packages/openapi3/test/xml-models.test.ts

```diff
@@ -0,0 +1,705 @@
+      properties: {
+        content: { type: "string" },
+      },
+      required: ["content"],
+      xml: {
+        name: "xmlBook",
+      },
+    });
+  });
+
+  it("set the element value for array property via @name", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @name("xmlBook")
+        model Book {        
+          tags: string[];
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        tags: {
+          type: "array",
+          items: {
+            type: "string",
+          },
+          xml: {
+            wrapped: true,
+          },
+        },
+      },
+      required: ["tags"],
+      xml: {
+        name: "xmlBook",
+      },
+    });
+  });
+
+  it("set the attribute value via @name", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `model Book {
+          @name("xmlcontent")
+          content: string;
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        content: { type: "string", xml: { name: "xmlcontent" } },
+      },
+      required: ["content"],
+    });
+  });
+
+  it("set the value on scalar via @name", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @name("xmlBook")
+        scalar Book extends string;`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "string",
+      xml: {
+        name: "xmlBook",
+      },
+    });
+  });
+
+  it("compare with the json name", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @name("xmlBook")
+        @encodedName("application/json", "jsonBook")
+        model Book {        
+          content: string;
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        content: { type: "string" },
+      },
+      required: ["content"],
+      xml: {
+        name: "xmlBook",
+      },
+    });
+  });
+
+  it("set the json name and no xml name", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @encodedName("application/json", "jsonBook")
+        model Book {        
+          content: string;
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        content: { type: "string" },
+      },
+      required: ["content"],
+    });
+  });
+});
+
+describe("@attribute", () => {
+  it("mark property as being an attribute", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `model Book {
+          @attribute
+          id: string;
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        id: { type: "string", xml: { attribute: true } },
+      },
+      required: ["id"],
+    });
+  });
+});
+
+describe("@unwrapped", () => {
+  it("provide the namespace and prefix using string", async () => {
```

Comment:
this case should emit a warning as its not supported by openapi 

https://github.com/OAI/OpenAPI-Specification/issues/630